pkgs/tools/admin/awscli2/default.nix

```
@@ -14,12 +14,14 @@ let
   py = python3 // {
     pkgs = python3.pkgs.overrideScope (final: prev: {
       ruamel-yaml = prev.ruamel-yaml.overridePythonAttrs (prev: {
+        version = "0.17.21";
         src = prev.src.override {
           version = "0.17.21";
           hash = "sha256-i3zml6LyEnUqNcGsQURx3BbEJMlXO+SSa1b/P10jt68=";
         };
       });
       urllib3 = prev.urllib3.overridePythonAttrs (prev: {
+        version = "1.26.18";
```

I'm getting this assertion on latest `nixos-unstable-small`:
```
        error: assertion '((pyproject != null) -> (format == null))' failed

        at /nix/store/izb28gwkf7i3l47nk7h86h34liq615ri-nixpkgs-patched-7bea27b/pkgs/development/interpreters/python/mk-python-derivation.nix:110:1:
           109|
           110| assert (pyproject != null) -> (format == null);
              | ^
           111|
```
May I suggest also adjusting the pyproject on the override?
```suggestion
        version = "1.26.18";
        pyproject = null;
```

---

Also, maybe a `rec` could help ensure the `version` property and the one inside `src.override` are the same?

---

> I'm getting this assertion on latest nixos-unstable-small:

Unfortunately I don't really understand the error message. When building awscli2 with the code from this PR, there is no error.

> Also, maybe a rec could help ensure the version property and the one inside src.override are the same?

Uh, yes, maybe. Is there some kind of "rec override"?

---

@andrevmatos @FliegendeWurst regarding that failed assertion
> <`@hexa:lossy.network`> set `pyproject = true;` and add `setuptools` into `nativeBuildInputs`

---

@FliegendeWurst the error message means that `mk-python-derivation` has set that, if `pyproject` is not null then `format` must be null; since `pyproject = true` in the up-to-date nixpkgs urllib3, when this is overriden here for the downgraded version and set `format = "setuptools"`, the assertion fails; You may not be getting this error yet because this happens only in latest nixpkgs master and nixos-unstable-small.

The fix, as I suggested, is to ensure `pyproject = null` in the override, so the `format = "setuptools"` is respected and the assertion doesn't fail.

About the `rec`, I don't know whay you mean by _rec override_, but just to give a concrete example, it should be enough to set something like this in this file:
```nix
      urllib3 = prev.urllib3.overridePythonAttrs (prev: rec {
        version = "1.26.18";
        format = "setuptools";
        pyproject = null;
        src = prev.src.override {
          inherit version;
          hash = "sha256-+OzBu6VmdBNFfFKauVW/jGe0XbeZ0VkGYmFxnjKFgKA=";
        };
      });
```

---

Ah okay. Will rebase and fix the error then. I'll try your override approach later today.

---

@andrevmatos The fix being proposed to solve the assertion issue is incorrect. Please check https://github.com/NixOS/nixpkgs/pull/254136. `format = "setuptools"` should be replaced with `pyproject = true` and `setuptools` must be added to `nativeBuildInputs`.

---

Also, the assertion issue is already being tackled on https://github.com/NixOS/nixpkgs/pull/268590. So not sure if you want to wait for it to be merged and then apply the version update to urllib3

---

#268590 has been merged

---

Sorry for the late response. Didn't get around to doing this after all. Perhaps someone else will pick it up eventually.